sdl-core/src/main/scala/io/smartdatalake/workflow/action/MLflowPredictAction.scala

```diff
@@ -0,0 +1,212 @@
+
+  private val transformerDefs: Seq[GenericDfsTransformer] = transformers
+
+  override val transformerSubFeedType: Option[Type] = {
+    val transformerTypeStats = transformerDefs
+      .map(_.getSubFeedSupportedType)
+      .filterNot(_ =:= typeOf[DataFrameSubFeed]) // ignore generic transformers
+      .groupBy(identity)
+      .mapValues(_.size)
+      .toSeq
+      .sortBy(_._2)
+    assert(
+      transformerTypeStats.size <= 1,
+      s"No common transformer subFeedType type found: ${transformerTypeStats
+        .map { case (tpe, cnt) => s"${tpe.typeSymbol.name}: $cnt" }
+        .mkString(",")}"
+    )
+    transformerTypeStats.map(_._1).headOption
+  }
+
+  validateConfig()
+
+  private def getTransformers(implicit context: ActionPipelineContext): Seq[GenericDfsTransformerDef] = {
+    mlflow.pythonMLflowApi.get.setLatest(mlflow.experimentId.get)
+
+    val getPredictionIdPython =
+      f"""
+         |import re
+         |
+         |class TrainingDataFrameError(Exception):
+         |  pass
+         |
+         |def get_predict_id(selector: str, ids: list[str]):
+         |  if len(ids) > 1:
+         |    # get DataObjectId according the given trainIdSelector
+         |    r = re.compile(".*${predictIdSelector.getOrElse("predict")}.*")
+         |    possible_predict_ids = list(filter(r.match, ids))
+         |    if len(possible_predict_ids) != 1:
+         |      raise TrainingDataFrameError(f"Zero or multiple DataObjectIds detected for prediction {possible_predict_ids}. One and only one DataObjectId must be provided for training. If only one Id is provided, this will be used. Otherwise the Id is selected using the case sensitive trainIdSelector (${predictIdSelector.get})")
+         |    return possible_predict_ids[0]
+         |  elif len(ids) == 1:
+         |    return ids[0]
+         |  else:
+         |    raise TrainingDataFrameError(f"Zero input DataObjectId provided {ids}.")
+         |""".stripMargin
+
+    // handle MLflow interaction in different phases
+    val transformer = context.phase match {
+      // return empty dataframe in Init phase
+      case ExecutionPhase.Init =>
+        val initMakePredictionsPython =
+          s"""
+             |import mlflow
+             |import json
+             |from pyspark.sql.types import StringType
+             |from pyspark.sql.functions import lit
+             |
+             |mlflow.set_tracking_uri("${mlflow.trackingURI}")
+             |# set experiment name
+             |mlflow.set_experiment("${mlflow.experimentName}")
+             |# get predict dataframe from SDL
+             |predict_id = get_predict_id("$predictIdSelector", list(inputDfs.keys()))
+             |print(f"The following DataObjectId will be used for training: {predict_id}")
+             |df = inputDfs[f"{predict_id}"]
+             |# add predictions; TODO: get correct type according to input
+             |df_predict = df.withColumn("predictions", lit(None).cast(StringType()))

```

Comment:
To allow proper validation in init phase, the model should already be loaded there. I expect that the downloaded model is cached, as there is a SparkModelCache in mlflow.
We could make a switch to disable loading the model in init phase for performance reasons.

Reply:
done.